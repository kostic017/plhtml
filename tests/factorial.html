<!doctype html>
<html lang="en">
    <head>
        <title>`Factorial`</title>
    </head>
    <body>

        <main>

            <var class="integer">quit</var>

            <div data-while="quit &ne; 1">

                <var class="integer">n</var>
                <var class="integer">i</var>
                <var class="integer">fact</var>

                <data value="1">i</data>
                <data value="1">fact</data>

                <output>`n: `</output>
                <input name="n">

                <div data-while="i &leq; n">
                    <data value="fact * i">fact</data>
                    <data value="i + 1">i</data>
                </div>

                <output>`Factorial of ` + n + ` is ` + fact</output>

                <output>`\nType 1 to quit: `</output>
                <input name="quit">

            </div>

        </main>

    </body>
</html>
