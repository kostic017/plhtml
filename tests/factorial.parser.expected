"Factorial"
Main:
    integer n
    integer i
    integer fact
    i = 1
    fact = 1
    Write: "n: "
    Read n
    While i LeqOp n
        fact = fact * i
        i = i + 1
    Write: "Factorial of " + n + " is " + fact
